### ISSUE TITLE
[QUIC] Participate in QUIC interop testing

### ISSUE THREAD

**Issue by scalablecory** [dotnet/runtime#43061]
Consider participating in automated QUIC interop testing to ensure basic compatibility with other implementations:

https://github.com/marten-seemann/quic-interop-runner/blob/master/README.md#building-a-quic-endpoint

**Comment by karelz** [comment]
Triage: We should build on top of what msquic has -- @nibanks do you have some existing interop tests for msquic?

**Comment by nibanks** [comment]
Yep. I recommend you start by porting: https://github.com/microsoft/msquic/blob/main/src/tools/interop/interop.cpp. Since you have HTTP/3 there are additional tests you can do beyond what MsQuic does.
